File: 02_automatedSimplifiedExtensionLayer/00_matlab/functions/get_process_adding.m
function [ add_manual_processes,adding_files ] = get_process_adding(path_alignments)

adding_files = struct2cell(dir([path_alignments]));
adding_files = adding_files(1,:)';



        counter=0;
        for i=3:length(adding_files)
            disp(adding_files{i})
            filename=fullfile(path_alignments,adding_files{i});
       
            % process names
            [~,~,process_adding.meta_data_processes] = xlsread(filename,'Process_meta_data');
            process_adding.meta_data_processes(find(cellfun(@(C)...
                any(isnan(C(:))), process_adding.meta_data_processes)))={[]};

            % A
            [process_adding.A.mean_values,~,~] = xlsread(filename,'SUMMARY A','D2:IA200');
            [types,process_adding.meta_data_flows,~] = xlsread(filename,'SUMMARY A','A2:C200');

            if ~isempty(process_adding.A.mean_values)
            process_adding.A.mean_values(isnan(process_adding.A.mean_values))=0;
            process_adding.A.mean_values(isempty(process_adding.A.mean_values))=0;
            end

            if ~isempty(types)
            process_adding.meta_data_flows(:,2)=num2cell(types);
            end

            % B
            [process_adding.B.mean_values,~,~] = xlsread(filename,'SUMMARY B','D2:IA500');
            [~,process_adding.meta_data_elementary_flows,~] = xlsread(filename,'SUMMARY B','A2:C500');

            if ~isempty(process_adding.B.mean_values)
            process_adding.B.mean_values(isnan(process_adding.B.mean_values))=0;
            process_adding.B.mean_values(isempty(process_adding.B.mean_values))=0;
            end

            % F
            [process_adding.F.mean_values,~,~] = xlsread(filename,'SUMMARY F','C2:IA200');
            [~,process_adding.meta_data_factor_requirements,~] = xlsread(filename,'SUMMARY F','A2:C200');

            if ~isempty(process_adding.F.mean_values)
            process_adding.F.mean_values(isnan(process_adding.F.mean_values))=0;
            process_adding.F.mean_values(isempty(process_adding.F.mean_values))=0;
            end
            
            counter=counter+1;
            add_manual_processes(counter)=process_adding;

        end
        
        
end

File: src/utils/get_process_adding.py
import os
import pandas as pd
import numpy as np

def get_process_adding(path_alignments):
    # List files in the specified directory
    adding_files = [f for f in os.listdir(path_alignments) if os.path.isfile(os.path.join(path_alignments, f))]

    add_manual_processes = []
    counter = 0

    for i in range(2, len(adding_files)):  # Start from index 2 (equivalent to MATLAB starting from 3)
        print(adding_files[i])
        filename = os.path.join(path_alignments, adding_files[i])

        process_adding = {}

        # Process names (meta_data_processes)
        process_adding['meta_data_processes'] = pd.read_excel(filename, sheet_name='Process_meta_data', header=None).values.tolist()
        process_adding['meta_data_processes'] = [[None if pd.isna(cell) else cell for cell in row] for row in process_adding['meta_data_processes']]

        # Sheet SUMMARY A
        process_adding['A'] = {}
        process_adding['A']['mean_values'] = pd.read_excel(filename, sheet_name='SUMMARY A', usecols="D:IA", skiprows=1, nrows=198).values
        types = pd.read_excel(filename, sheet_name='SUMMARY A', usecols="A:C", skiprows=1, nrows=198).values
        process_adding['meta_data_flows'] = pd.read_excel(filename, sheet_name='SUMMARY A', usecols="A:C", skiprows=1, nrows=198).values.tolist()

        if process_adding['A']['mean_values'].size > 0:
            process_adding['A']['mean_values'] = np.nan_to_num(process_adding['A']['mean_values'], nan=0.0)

        if len(types) > 0:
            for idx, val in enumerate(types):
                process_adding['meta_data_flows'][idx][1] = val

        # Sheet SUMMARY B
        process_adding['B'] = {}
        process_adding['B']['mean_values'] = pd.read_excel(filename, sheet_name='SUMMARY B', usecols="D:IA", skiprows=1, nrows=498).values
        process_adding['meta_data_elementary_flows'] = pd.read_excel(filename, sheet_name='SUMMARY B', usecols="A:C", skiprows=1, nrows=498).values.tolist()

        if process_adding['B']['mean_values'].size > 0:
            process_adding['B']['mean_values'] = np.nan_to_num(process_adding['B']['mean_values'], nan=0.0)

        # Sheet SUMMARY F
        process_adding['F'] = {}
        process_adding['F']['mean_values'] = pd.read_excel(filename, sheet_name='SUMMARY F', usecols="C:IA", skiprows=1, nrows=198).values
        process_adding['meta_data_factor_requirements'] = pd.read_excel(filename, sheet_name='SUMMARY F', usecols="A:C", skiprows=1, nrows=198).values.tolist()

        if process_adding['F']['mean_values'].size > 0:
            process_adding['F']['mean_values'] = np.nan_to_num(process_adding['F']['mean_values'], nan=0.0)

        # Add the processed data to the list
        counter += 1
        add_manual_processes.append(process_adding)

    return add_manual_processes, adding_files
